launcher/ui/dialogs/ModDownloadDialog.cpp

```diff
@@ -83,16 +115,38 @@ QList<BasePage *> ModDownloadDialog::getPages()
     };
 }
 
-void ModDownloadDialog::setSuggestedMod(const QString& name, ModDownloadTask* task)
+void ModDownloadDialog::addSelectedMod(const QString& name, ModDownloadTask* task)
+{
+    if(modTask.contains(name))
+        delete modTask.find(name).value();
```

Comment:
nit: Maybe we can just call `removeSelectedMod` here?

Reply:
I was trying to avoid calling `modTask.remove()` where possible because I imagine there is a possibility of it generating a rehash, although I don't know the exact internal structure of it to be certain... If it helps readability though, I'll do that.